### ISSUE TITLE
Differentiate axis label from category in CategoryAxis?

### ISSUE THREAD

**Issue by brecriffs** [amcharts/amcharts4#1904]
Let's say my data is:
```
[{
  "id": "ksahfu164598",
  "name": "Jim",
  "value": 3.5
}, {
  "id": "ampsha15647",
  "name": "Jim",
  "value": 6
}, {
  "id": "jdjama155835",
  "name": "Billy",
  "value": 6.5
}]
```
and the category axis is defined as:
```
let categoryAxis = chart.xAxes.push(new am4charts.CategoryAxis());
categoryAxis.dataFields.category = "id";
categoryAxis.dataFields.axisLabel = "name";
```

I want the categories for the 2 'Jim' entries to be separate, based on the id. But I still would like to keep 'Jim' as the axis label. Is this possible, by setting a dataField of 'axisLabel' or something?

I know I can use an adapter to edit the category on the fly, but it will be messier. My categories will not always be a user id, sometimes they are something else.

**Comment by martynasma** [comment]
I don't think there's any better way to handle repeating categories than using an adapter :/

https://www.amcharts.com/docs/v4/tutorials/handling-repeating-categories-on-category-axis/

**Comment by brecriffs** [comment]
Alright, thanks anyways!

**Comment by jbate** [comment]
This issue and the associated tutorial (thanks @martynasma!) helped me out with a similar problem. Although the solution wasn't that satisfactory (as I had to iterate over the data I'd ingested in and pad it with a fake suffix, and it made the data look a bit "ugly"), it did the job!

Top tip for anyone else coming here though... (and maybe the tutorial can explicitly mention this?), you'll also have to run the same regex over your tooltips.

Is there a way to create a custom text formatter that I can use to run the regex function? (I ended up using an adapter on the tooltip label's `textOutput`).
